torch/nn/parallel/distributed.py

```diff
@@ -1385,9 +1433,22 @@ def _lazy_init(self):
         self._setup_in_backward_optimizers()
         self._lazy_init_ran = True
 
+    def _should_disable_reducer(self) -> bool:
+        return torch._utils.is_compiling() or self._force_to_disable_reducer
+
     def _pre_forward(self, *inputs, **kwargs):
-        if not self._lazy_init_ran:
+        if self._ddp_python_hook and self._should_disable_reducer():
```

Comment:
why is `self._ddp_python_hook` checked _outside_ of _should_disable_reducer, instead of putting the and condition _inside_ ? it seems like they are used 1:1